sdk/nodejs/index.ts

```diff
@@ -0,0 +1,489 @@
+        return input.length;
+    } else {
+        return Object.keys(input).length;
+    }
+}
+
+export function list<T>(...elements: T[]): T[] {
+    return elements;
+}
+
+export function log(x: number, base: number): number {
+    return Math.log(x) / Math.log(base);
+}
+
+export function lookup<T>(inputMap: { [k: string]: T }, key: string, defaultValue?: T): T {
+    const val = inputMap[key];
+    if (val === undefined) {
+        if (defaultValue === undefined) {
+            throw new Error("Cannot find key '" + key + "' in map");
+        } else {
+            return defaultValue;
+        }
+    }
+
+    return val;
+}
+
+export function lower(input: string): string {
+    return input.toLowerCase();
+}
+
+export function map(...keyValuePairs: any[]): { [k: string]: any } {
+    if (keyValuePairs.length % 2 !== 0) {
+        throw new Error("Number of parameters to map must be even");
+    }
+
+    const result: { [k: string]: any } = {};
+
+    for (let i = 0; i < keyValuePairs.length; i += 2) {
+        const key = keyValuePairs[i];
+        result[key] = keyValuePairs[i + 1];
+    }
+
+    return result;
+}
+
+export function matchkeys() {
+    // TODO(jen20)
+}
+
+export function max(...numbers: number[]): number {
+    return Math.max(...numbers);
+}
+
+export function merge(...maps: { [k: string]: any }[]): { [k: string]: any } {
+    const output = maps[0];
+    for (const x of maps) {
+        for (const y in x) {
+            if (x.hasOwnProperty(y)) {
+                output[y] = x[y];
+            }
+        }
+    }
+
+    return output;
+}
+
+export function min(...numbers: number[]): number {
+    return Math.min(...numbers);
+}
+
+export function md5(input: string): string {
+    const hash = crypto.createHash("md5");
+    hash.update(input);
+    return hash.digest().toString("hex");
+}
+
+export function pathexpand(path: string): string {
+    const home = os.homedir();
+    return path.replace(/^~(?=$|\/|\\)/, home);
+}
+
+export function pow(base: number, exponent: number): number {
+    return Math.pow(base, exponent);
+}
+
+export function sha1(input: string): string {
+    const hash = crypto.createHash("sha1");
+    hash.update(input);
+    return hash.digest().toString("hex");
+}
+
+export function sha256(input: string): string {
+    const hash = crypto.createHash("sha256");
+    hash.update(input);
+    return hash.digest().toString("hex");
+}
+
+export function sha512(input: string): string {
+    const hash = crypto.createHash("sha512");
+    hash.update(input);
+    return hash.digest().toString("hex");
+}
+
+export function signum(input: number): number {
+    if (input === 0) {
+        return 0;
+    }
+    if (input < 0) {
+        return -1;
+    } else {
+        return 1;
+    }
+}
+
+export function upper(input: string): string {
+    return input.toUpperCase();
+}
+
+export function transpose(inputMap: { [k: string]: Array<string> }): { [k: string]: string[] } {
```

Comment:
Any reason for `Array<string>` vs. `string[]` in the signature here (they should be the same, and the latter is also used later in the signature.

Reply:
:+1: